main.js

```diff
@@ -1048,21 +1061,58 @@ Creep.prototype.travelToRoom = function travelToRoom(tgtRoom, forward) {
 		return ERR_NO_PATH;
 
 	let list_route = _.get(this, ["memory", "list_route"]);
-	if (list_route != null) {
-		if (forward == true) {
+	
+	if (list_route && list_route.length > 0) {
+		if (forward === true) {
 			for (let i = 1; i < list_route.length; i++) {
-				if (_.get(list_route, i) != null && this.room.name == _.get(list_route, i - 1)) {
+				if (this.room.name === list_route[i - 1]) {
+					// Check for portal to next room in the route
+					let portal = this.pos.findClosestByRange(FIND_STRUCTURES, {
```

Comment:
Note for future reference- this is probably a CPU-expensive routine to run every tick. Can be optimized when needed with caching.